pkg/provision/source.go

```diff
@@ -30,6 +30,13 @@ type ReservationPoller interface {
 	Poll(nodeID pkg.Identifier, from uint64) (reservations []*Reservation, lastID uint64, err error)
 }
 
+// ReservationFromSource wraps a reservation type and has
+// a boolean to indicate it is the last reservation
+type ReservationFromSource struct {
+	*Reservation
```

Comment:
I don't think you need to have a pointer here. You already use a pointer of the ReservationFromSource in the channel.